decidim-pages/lib/decidim/pages/feature.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+Decidim.register_feature(:pages) do |feature|
+  feature.component :page do |component|
+    component.engine = Decidim::Pages::Engine
+    component.admin_engine = Decidim::Pages::AdminEngine
+
+    component.on(:create) do |instance|
+      Decidim::Pages::CreatePage.call(instance) do
+        on(:error) { raise "Can't create page" }
+      end
+    end
+
+    component.on(:destroy) do |instance|
+      Decidim::Pages::DestroyPage.call(instance) do
+        on(:error) { raise "Can't destroy page" }
```

Comment:
This is for internal use and won't be leaked into the interface, no need to I18n.